config/nginx.conf.erb

```diff
@@ -37,45 +39,74 @@ http {
     port_in_redirect off;
 
     more_clear_headers 'X-Powered-By';
-    add_header X-Frame-Options DENY always;
-    add_header X-Content-Type-Options nosniff always;
-    add_header X-XSS-Protection "1; mode=block" always;
-    add_header Strict-Transport-Security max-age=31536000 always;
-    add_header Content-Security-Policy "
-     default-src 'none';
-     base-uri 'self';
-     form-action 'none';
-     object-src 'none';
-     connect-src 'self';
-     font-src 'self';
-     script-src 'self';
-     img-src 'self' https://*.githubusercontent.com/u/;
-     style-src 'self' 'unsafe-inline'" always;
+
+    proxy_http_version 1.1;
+    proxy_set_header Upgrade $http_upgrade;
+    proxy_set_header Connection 'upgrade';
+    proxy_set_header Host $host;
+    proxy_cache_bypass $http_upgrade;
 
     if ($http_x_forwarded_proto != 'https') {
       rewrite ^ https://$host$request_uri? permanent;
     }
 
-    location / {
-      root build/;
-      try_files $uri /index.html =404;
+    location /_next/static {
+      # proxy_cache STATIC;
+      proxy_pass http://nextjs_upstream;
+
+      # For testing cache - remove before deploying to production
+      # add_header X-Cache-Status $upstream_cache_status;
+
+      # CSP
+      add_header X-Frame-Options DENY always;
```

Comment:
A follow-up will be to externalize these as includes to avoid duplication. Or another idea might be to use https://github.com/trezy/next-safe/ and serve these from the app.